sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php

```diff
@@ -102,6 +102,27 @@ function get_uptime_stats() {
 	return($status);
 }
 
+/* Returns CPU temperature if available from the system */
+function get_cpu_temperature() {
```

Comment:
There are very similar function in the main repo under https://github.com/pfsense/pfsense/blob/master/src/usr/local/www/includes/functions.inc.php#L141 and https://github.com/pfsense/pfsense/blob/master/src/usr/local/www/widgets/include/thermal_sensors.inc#L33 -- It would be best to include (not copy) and use one of those functions instead of making a new one that does essentially the same task.

Reply:
Hi Jim, thank you for looking into this.

If I try and include those, such as functions.inc.php to use get_temp(), it won't allow it since there's similar named functions in lcdproc_client.php. First error: `PHP ERROR: Type: 64, File: /usr/local/www/includes/functions.inc.php, Line: 84, Message: Cannot declare cpu_usage() (previously declared in /usr/local/pkg/lcdproc_client.php:69)` 

My PHP is rusty but I had a look and it seems to not be like Python where I can import specific functions. All or nothing if I'm not mistaken? There's also a lot of other same named functions in lcdproc_client.php that would clash.

So if this leaves me with my original code and how it's done here, I've also just about got a commit to switch between celsius and fahrenheit.

Cheers.

Reply:
It's likely that those functions also overlap and work in a similar manner. You can probably use the one from functions.inc or rename the lcdproc function -- ideally the lcdproc-specific functions should all be named something unique anyhow, perhaps stating with `lcdproc_<name>`.  PHP namespaces are also possible but more complicated. It's best to avoid the conflict entirely and remove as much redundant code as possible.

Reply:
PHP namespaces looks very interesting. That seems like the ideal route if I'm not mistaken. While I do see it's not ideal to have the other redundant code, that's a bit of tech debt that eventually needs to be sorted. But at least I can start with what I implement using better practices.

So, before dinner a quick initial toe in the water:
`namespace LcdProc;` and `require_once("includes/functions.inc.php");` in lcdproc_clients.php

That cleared up any function collisions and allowed me to use get_temp() in my get_cputemperature() function. That worked and returned what I expected.

Just some validation before I continue on. Is this the correct path with my initial idea above?

Reply:
It allows it to function but it ignores the underlying problem of similar functions duplicating code and effort. IMO, it's the wrong solution

Reply:
What would be the ideal solution and how should I go about it?